apps/prairielearn/src/server.js

```diff
@@ -1405,6 +1405,7 @@ module.exports.initExpress = function () {
   // which checks the assessment type and calls next() if it's not the right type
   app.use('/pl/course_instance/:course_instance_id/assessment/:assessment_id', [
     require('./middlewares/selectAndAuthzAssessment'),
+    require('./middlewares/clientFingerprint').default,
```

Comment:
Question for @mwest1066: while reviewing this, I noticed that we log page views before checking student access - is that intentional? Do we want the same for fingerprints?

Reply:
Huh, that feels backwards to me. I would have thought we'd always do access checks before anything else.

Reply:
Sorry, I might be misunderstanding the question or the code. It looks to me like we currently:
1. Perform authorization checks
2. Create and update fingerprints in the DB
3. Log the page view (with an entry in `page_view_logs`, as distinct from earlier logging of the request)

This looks like the correct order to me? It does not log the fingerprint change for students who can't access the exam? Or have I misunderstood something here?

Reply:
There's a distinction between authorization and access checks (the `studentAssessmentAccess` middleware a few lines below this). That middleware had the power to 403 the response. My concern is recording fingerprints and logs for a response that was 403ed (that is, the student couldn't access what they were trying to).

Reply:
Thank you! I was forgetting about `studentAssessmentAccess` and focusing only on `selectAndAuthzAssessmentInstance`. As far as I can tell, `studentAssessmentAccess` only blocks access or redirects, so we don't care about logging this. I agree we should move `studentAssessmentAccess` above `clientFingerprint`.

Reply:
I opened #8932 to address this.

Reply:
@mylesw27 you'll have to rebase this PR once that lands and ensure that the client fingerprint middleware are ordered correctly (that is, after the `studentAssessmentAccess` middleware).